### ISSUE TITLE
Unknown command 'listpays'

### ISSUE THREAD

**Issue by sumBTC** [ElementsProject/lightning#3931]
### Issue and Steps to Reproduce
Sometimes (this happened a few times now) and maybe when interacting with `spark-wallet`, `c-lightning` loses `listpays` and responds:

```
{
   "code": -32601,
   "message": "Unknown command 'listpays'"
}

```
After a restart it knows about `listpays` again :-)

### `getinfo` output

version: "v0.9.0-2-ge7d89cd"

**Comment by vincenzopalazzo** [comment]
The plugin crashed for some reason, can you share your log?

**Comment by sumBTC** [comment]
Oh you are right but I restarted c-lightning again and the log is gone (output of `lightningd`). Stupid me! I will include it when it happens again.

**Comment by vincenzopalazzo** [comment]
:smile: Yep don't worry. I suggest (if you can) you to use the next release of lightningd (9.0.1) because include some refactoring inside the listpays, like https://github.com/ElementsProject/lightning/issues/3915 but add also a little refactoring inside the format of the response more here https://github.com/ElementsProject/lightning/pull/3888.

**Comment by sumBTC** [comment]
I will but only when `spark-wallet` is also up-to-date.

**Comment by ZmnSCPxj** [comment]
PR #3890 should also crash `lightningd` if any of the builtin plugins crash, which should help, if you are willing to run on master and possibly have C-lightning crash outright.

**Comment by sumBTC** [comment]
I'm now on master and `spark-wallet` is still fine.

**Comment by cdecker** [comment]
I'm pretty sure this is just another instance of #3915, which will be fixed by #3927 :+1:

**Comment by sumBTC** [comment]
Ok I close it for now
